### PR TITLE
[develop] Update GitHub Actions workflows to run on specific branches

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
## DESCRIPTION OF CHANGES: 

GitHub Actions workflows keep trying to run on my forks when I push new branches. Many times I'm not ready to run the tests when I'm backing up my branches.

I think it's a better use of resources to run those tests when a PR is open, and when that PR is merged, or a push directly to a set of specific branches occurs. The goal of this code change the behavior of the GHA workflows to do just that.

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

These changes aren't relevant to anything other than the GHA test workflows, and can only be tested via this PR. Fingers crossed!

- [ ] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

### CODE REVIEW

File: .github/workflows/python_unittests.yaml [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+on:
+  push:
+    branches:
+      - develop
+      - 'release/*'
+  pull_request:
+    branches:
+      - develop
+      - 'release/*'
+  workflow_dispatch:
```
Comment (christinaholtNOAA) [inline .github/workflows/python_unittests.yaml:11]: Might as well add a manual trigger to this one, too.

**Comment by danielabdi-noaa** [comment]
@christinaholtNOAA Can we remove `build.yml` with this PR while you are at it? It is a self hosted runner on cheyenne that never seems to run. I get a ton of emails of failed builds because of it, and it is not useful anymore  now that there is Jenkins.

**Comment by christinaholtNOAA** [comment]
@danielabdi-noaa I get a bunch of those, too. I'm curious what others think. It looks like @mark-a-potts and @willmayfield were involved in introducing that file, so I'd like to get their input on its removal. 

I'm happy to make the change, if we decide on it.

**Comment by mark-a-potts** [comment]
It is fine with me to remove it.

**Review by danielabdi-noaa — APPROVED** [review]

**Review by MichaelLueken — APPROVED** [review]

**Pull request merged**